server/src/main/java/io/crate/execution/engine/collect/sources/InformationSchemaIterables.java

```diff
@@ -214,18 +215,20 @@ private PgClassTable.Entry relationToPgClassEntry(RelationInfo info) {
             info.ident().name(),
             toEntryType(info.relationType()),
             info.columns().size(),
-            info.primaryKey().size() > 0);
+            !info.primaryKey().isEmpty());
     }
 
     private PgClassTable.Entry primaryKeyToPgClassEntry(RelationInfo info) {
         return new PgClassTable.Entry(
             Regclass.primaryOid(info),
             OidHash.schemaOid(info.ident().schema()),
             info.ident(),
-            info.ident().name() + "_pkey",
+            info.pkConstraintName() != null
+                ? info.pkConstraintName()
+                : info.ident().name() + "_pkey",
```

Comment:
Should not this be `PK_SUFFIX` (_pk) to fully address #17377 ?

Reply:
This is potentially a breaking change, and it most probably need to be the other way around, so change `key_column_usage`, to use `_pkey`, since this is what postgres does for both tables.

Reply:
Ok, but 17377 should stay open then

Reply:
The issue, only describes the problem that although there is a custom pk name, the default is used. This PR fixes the issue.

Reply:
True, it fixes only the 1st part. The 2nd in my opinion is a breaking change and cannot be backported to hotfix releases.

Reply:
After internal discussion we decided to change to `_pkey` and treat this as a fix which will also backport.
Added a separate commit which deals with this.